src/services/formatting/formatting.ts

```diff
@@ -432,13 +432,12 @@ namespace ts.formatting {
             if (leadingTrivia) {
                 indentTriviaItems(leadingTrivia, indentation, /*indentNextTokenOrTrivia*/ false,

```

Comment:
how does `indentTriviaItems` already get rid of trailing whitespace for trivia?

Reply:
If there's leading trivia on the node, we first parse those in `indentTriviaItems`. This method pretty much apply rules for trivia only (i.e. indenting multiline/single line). The later `trimTrailingWhitespacesForRemainingRange`, used to ignore this trivia and cause some changes to conflict between them.